### ISSUE TITLE
Merge Populations

### ISSUE THREAD

**Issue by msevestre** [Open-Systems-Pharmacology/PK-Sim#2613]
How can we implement this feature? This is not as easy as I thought. 

* User can select multiple populations to merge
* A view will be shown listing the selected population as well as the base individual that will be used for this
* Why do we need a base individual? The individual is used to determine the value of all parameters that are not aged-dependent. 
   * => That means that merging populations using completely different   base individuals would result in BS (say I update ph of one organ in one individual but not in the other and then create two populations, this would get out of sync)
   * only common expression profiles will be added automatically. Otherwise, they will be removed
   * Expression cannot be added to the merge population
   * User-defined variability CANNOT BE DEFINED HERE. 
   * Reference to population will be saved in the merged population => This is required for snapshot creation; otherwise, we won't be able to recreate it. 

So, in short, a MergePopulation will only combine the existing population, but we won't allow editing much of its property. WHY, so that we can recreate from snapshot. 

Feedback @PavelBal @StephanSchaller @Yuri05

**Comment by msevestre** [comment]
PS
why do we need this? because we will create 3 different population for HI (Score A to C) and we can combine them that way

**Comment by Yuri05** [comment]
>only common expression profiles will be added automatically. Otherwise, they will be removed

I think we should not remove expression profiles, because this would change the existing population. So I would suggest to **stop** merging in such a case and **suggest to the user to create new const Zero expression Profiles and add them to all populations where the corresponding proteins are missing**; after that the merging could be performed.

**Comment by Yuri05** [comment]
>A view will be shown listing the selected population as well as the base individual that will be used for this 

I wonder if we should only allow merging of populations which where **created based on the same individual**.

My expectation is always: merging populations does not change any of the population properties.
But if the original populations was created based on individual I1, and my merged population is suddenly created based on the individual I2 - this would change some properties of the individuals within the original population - which would be completely wrong in my opinion.

So I would suggest that in the merge populations view no base individual has to be selected.
Instead, it should be checked if all selected populations are based on the same individual. 
If yes - OK to merge. If no - merging not possible.

**Comment by msevestre** [comment]
so unless I am mistaken, we need to be able to use different individuals
If someones changes a population after the fact, this will also create problems. Like adding an expression profile or removing it etc...

**Comment by Yuri05** [comment]
>I'd love to say it's a copy. That would be much easier. But then no snapshot...

Everything else (besides the snapshot) can be achieved already now by merging via CSV export/import.
And in the "csv-merged" population we can also add/modify user defined variability, add expressions etc.

So if it's a copy: then I would suggest to save the snapshots of all input populations as part of the merged population and export all of them into the project snapshot. Then the merged population still can be recretaed even if the input populations were changed after merge.

(By the way: if a user defined variabilty was added in one of the input populations: it's displayed as "unknown"  distribution in the "csv-merged" population, which is quite nice. I never tried it before :)
![grafik](https://user-images.githubusercontent.com/25061876/236228268-ef405bcd-1ea5-4217-be80-611b071e3d5b.png)

**Comment by Yuri05** [comment]
A completely different approach to think about: instead of merging populations - allow selection of **multiple** populations when creating a simulation (with the same restrictions as discussed above: e.g. only populations with the same expression profiles can be selected etc.)

**Comment by msevestre** [comment]
>So if it's a copy: then I would suggest to save the snapshots of all input populations as part of the merged population and export all of them into the project snapshot. Then the merged population still can be recretaed even if the input populations were changed after merge.

this simple features becomes way too complicated. Right now,we need to simplify or we won't ever ship

**Comment by Yuri05** [comment]
The simplest solution: create a "csv-merged" population.
Thus in the background just export all input populations to temporary csv and reimport them into one merged pop in the same way as it's currently done for the pop csv import.
The only gain: user does not need to export/import populations by hand.

**Comment by msevestre** [comment]
I still believe there is a case to be made for a reference only to populations that is restricted in some ways.
Or we do indeed the same as CSV import .We lose completely the connection. We are on the first individual only 
For snapshot, we could theoretically also export the pop as csv relative to the snapshot file and reference it in the snapshot//

**Comment by msevestre** [comment]
I have been thinking about it and I think merged population is not a good idea after all
What we could use, is a way to split population results instead
So here
![image](https://user-images.githubusercontent.com/1041237/236355794-2f6a7e28-02cd-40ca-a88c-52db9d1a3e87.png)

What could be done is simply to allow for proportions of populations. I have 3 simulations, say HI A, HI B, and HIC
And I want to reproduce a study with 10% A, 30%B, 60%C

We can add another column to this view (default is 100%) where we specify the proportion of EACH that we want to take
In the Comparison, we would need to save this percentage, as well as the id of all individuals selected randomly (if I have 100 ind and I take 10%, we would save 10 ids). 

This would allow us to merge populations OUTPUT. That way we do not care if they have the same user variability, enzymes etc..

**Comment by msevestre** [comment]
So we need 2 more columns (Percentage, editable) and (Count = total * percentagE)
Where we have the name we could add the number of ind as well pop1 (n = 1500)
